src/common/reduction/matching/GapMatcher.cpp

```diff
@@ -63,3 +70,121 @@ std::string GapMatcher::config(const std::string &prepend) const {
   ss << prepend << fmt::format("minimum_time_gap: {}\n", minimum_time_gap_);
   return ss.str();
 }
+
+void GapMatcher::split_and_stash_event(Event evt) {
+  Cluster new_cluster;
+  std::vector<Cluster> new_clusters_a = split_cluster(evt.ClusterA);
+  std::vector<Cluster> new_clusters_b = split_cluster(evt.ClusterB);
+  std::vector<Event> new_events;
+
+  for (Cluster cluster_a : new_clusters_a) {
```

Comment:
TODO, ask francesco/find stats on the freia multiple events per time window, consider changing out to double hits instead of unlimited